crates/solver/src/solver/score_computation.rs

```diff
@@ -0,0 +1,135 @@
+// Suppose a solver generates a solution, and we compute the three variables:
+// surplus, fees and gas_cost. Currently, the objective is then simply equal to
+// surplus + fees - gas_cost. However, for CIP-20, we now have one additional
+// step for each solver. So for each solver we have trained a revert risk model,
+// which essentially is a function that computes the probability of the
+// settlement not reverting, and is a function of 3 parameters.
+// p (num_orders_in_sol, gas_used, gas_price).
+// So we compute this p, and then the score is equal to:
+// score = p * (surplus + fees) - gas_cost
+
+use {
+    super::SolverType,
+    crate::objective_value,
+    anyhow::{Context, Result},
+    clap::{Parser, ValueEnum},
+    ethcontract::U256,
+    num::ToPrimitive,
+    std::{
+        collections::HashMap,
+        fmt::{Display, Formatter},
+        ops::Neg,
+        str::FromStr,
+    },
+};
+
+#[derive(Debug, Default, Clone)]
+pub struct ScoreCalculator {
+    a: f64,
+    b: f64,
+    c: f64,
+    x: f64,
+}
+
+impl ScoreCalculator {
+    pub fn new(a: f64, b: f64, c: f64, x: f64) -> Self {
+        Self { a, b, c, x }
+    }
+
+    pub fn calculate(&self, inputs: objective_value::Inputs, nmb_orders: usize) -> Option<U256> {
+        let surplus = inputs.surplus_given.to_f64()?;
+        let fees = inputs.solver_fees.to_f64()?;
+        let gas_amount = inputs.gas_amount.to_f64()?;
+        let gas_price = inputs.gas_price.to_f64()?;
+
+        let exponent =
+            self.x.neg() - self.a * gas_amount - self.b * gas_price - self.c * nmb_orders as f64;
```

Comment:
It's a bit surprising that all these values get negated. Is there a reason to do it like this instead of passing the already negated parameters via the CLI?

Reply:
Hm, it's probably because the original formula that @harisang came up with make more sense using the `-`. TBH I did not get much into the formula itself.

Reply:
Fair point, @MartinquaXD . This is usually how logistic regression is formulated, and this is the only reason the parameters were computed like this. Of course what you propose is completely equivalent.

![image](https://user-images.githubusercontent.com/64154020/223076613-9ff75fd7-84bd-471d-ada7-ba8d04e5c614.png)

Reply:
Since it's usually done like this in math I think it's fine to leave it as is.